src/app/utils/useQueue.tsx

```diff
@@ -1,11 +1,12 @@
-import { useState } from "react";
+import { useState } from 'react';
 
 export const useQueue = <T,>() => {
   const [queue, setQueue] = useState<T[]>([]);
 
   const first = queue[0];
-  const length = queue.length;
+  const { length } = queue;
   const enqueue = (item: T) => setQueue((prevState) => [...prevState, item]);
+  // eslint-disable-next-line @typescript-eslint/no-unused-vars
   const dequeue = () => setQueue(([_, ...rest]) => [...rest]);
```

Comment:
You can remove the `_` here to get rid of `no-unsed-vars` error.

```suggestion
  const dequeue = () => setQueue(([, ...rest]) => [...rest]);
```